repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java

```diff
@@ -1690,6 +1687,92 @@ private EntityMutationContext preCreateOrUpdate(EntityStream entityStream, Entit
         return context;
     }
 
+    private void autoUpdateStarredDetailsAttributes(AtlasEntity entity, AtlasVertex vertex) {
+        Boolean starEntityForUser = entity.getStarred();
+
+        if (starEntityForUser != null) {
+
+            long requestTime = RequestContext.get().getRequestTime();
+            String requestUser = RequestContext.get().getUser();
+
+            Set<String> starredBy = new HashSet<>();
+            Set<AtlasStruct> starredDetailsList = new HashSet<>();
+            int starredCount = 0;
+
+            if (vertex != null) {
+                Set<String> vertexStarredBy = vertex.getMultiValuedSetProperty(ATTR_STARRED_BY, String.class);
+                if (vertexStarredBy != null) {
+                    starredBy = vertexStarredBy;
+                }
+
+                Iterable<AtlasEdge> starredDetailsEdges = vertex.getEdges(AtlasEdgeDirection.OUT, "__" + ATTR_STARRED_DETAILS_LIST);
+                for (AtlasEdge starredDetailsEdge : starredDetailsEdges) {
+                    AtlasVertex starredDetailsVertex = starredDetailsEdge.getInVertex();
+                    String assetStarredBy = starredDetailsVertex.getProperty(ATTR_ASSET_STARRED_BY, String.class);
+                    Long assetStarredAt = starredDetailsVertex.getProperty(ATTR_ASSET_STARRED_AT, Long.class);
+                    AtlasStruct starredDetails = getStarredDetailsStruct(assetStarredBy, assetStarredAt);
+                    starredDetailsList.add(starredDetails);
+                }
+
+                starredCount = starredBy.size();
+            }
+
+            if (starEntityForUser) {
+
+                //Check and update starredBy Attribute
+                if (!starredBy.contains(requestUser)){
+                    starredBy.add(requestUser);
+                }
+
+                //Check and update starredDetailsList Attribute
+                boolean isStarredDetailsListUpdated = false;
+                for (AtlasStruct starredDetails : starredDetailsList) {
+                    String assetStarredBy = (String) starredDetails.getAttribute(ATTR_ASSET_STARRED_BY);
+                    if (assetStarredBy.equals(requestUser)) {
+                        starredDetails.setAttribute(ATTR_ASSET_STARRED_AT, requestTime);
+                        isStarredDetailsListUpdated = true;
+                        break;
+                    }
+                }
+                if (!isStarredDetailsListUpdated) {
+                    AtlasStruct starredDetails = getStarredDetailsStruct(requestUser, requestTime);
+                    starredDetailsList.add(starredDetails);
+                }
+
+            } else {
+
```

Comment:
Can we make few functions that will basically responsible for removing and adding stars, and if possible lets keep star related utility fuctions in separate class. Better for tracing down later

Reply:
Like for e.g `addStarredUser` and `removeStarredUser` like this, would be reusable later and easy to modify